cron/data/iterator.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Vershandlern 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permisshandlerns and
+// limitathandlerns under the License.
+
+package data
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/jszwec/csvutil"
+	"github.com/ossf/scorecard/repos"
+)
+
+type Iterator interface {
+	HasNext() bool
+	Next() (repos.RepoURL, error)
+}
+
+func MakeIterator(reader io.Reader) (Iterator, error) {
+	dec, err := csvutil.NewDecoder(csv.NewReader(reader))
+	if err != nil {
+		return &csvReader{}, fmt.Errorf("error in csvutil.NewDecoder: %w", err)
+	}
+	return &csvReader{decoder: dec}, nil
+}
+
+type inputRepo struct {
+	Repo     string `csv:"repo"`
+	Metadata string `csv:"metadata"`
+}
+
+type csvReader struct {
+	decoder *csvutil.Decoder
+	err     error
+	next    inputRepo
+}
+
+func (reader *csvReader) HasNext() bool {
+	reader.err = reader.decoder.Decode(&reader.next)
+	return reader.err != io.EOF
+}
+
+func (reader *csvReader) Next() (repos.RepoURL, error) {
+	if reader.err != nil {
+		return repos.RepoURL{}, reader.err
+	}
+	ret := repos.RepoURL{}
+	var err error
+	err = ret.Set(reader.next.Repo)
```

Comment:
nit: just `err := ` ?

Reply:
Go complains with `no new variables on left side of :=` if I use `err := `. Maybe I'm missing some Go syntax here?